### DISCUSSION TITLE
Updating for new versions of MacOS

### DISCUSSION THREAD

**Discussion by naikrovek** [Code-Hex/vz#160] [General]
How do you create the `.h` and `.m` files for new versions of MacOS?  I'm curious how to do this in case you ever choose to stop maintaining this.  These files may come with Xcode or MacOS somehow, I don't know.  I'm new to MacOS and Xcode.

Thank you

[added later] I *really* like this repo, or at least what I believe it to be at this time.  I'm currently using the last GPLv3 commit of cirruslabs/tart to do what I want, and I think this library will help me replace that tool with a Go program.  I just want to make sure that it will be possible to keep up with MacOS releases when required.

**Comment by cfergeau** [comment]
I believe these files are hand-written, the existing `virtualization_1*.[hm]` files have `created by codehex` at the beginning, https://github.com/Code-Hex/vz/pull/159/files adds new ones for macos 15, ...

**Reply by naikrovek** [reply]
Thank you!

**Comment by naikrovek** [comment]
if this is a stupid question, please let me know.  I'm new to MacOS and I've used Xcode maybe 5 times over the past 18 months for a grand total of 3-5 minutes combined.

**Comment by cfergeau** [comment]
If you are looking for command line tools based on Code-Hex/vz to start virtual machines, you can take a look at vfkit, lima, and there might be others.
